2 bug fixes

nsoranzo opened the pull request:
- Fix 2 param issues in `lib/galaxy/tools/duplicate_file_to_collection.xml`
- Fix traceback when exporting a workflow as BioCompute Object

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

nsoranzo committed:
Fix 2 param issues in ``duplicate_file_to_collection.xml``

Fix the following 2 errors produced when loading the tool:

```
ValueError: parameter 'input' uses an unknown type 'dataset'
```

and

```
galaxy.tools.parameters.basic.ParameterValueError: parameter 'number': the attribute 'value' must be set for non optional parameters
```

nsoranzo committed:
Small refactoring

nsoranzo committed:
Fix traceback when exporting a workflow as BioCompute Object

Fallback to tool id if a workflow step doesn't have a label. Fix:

```
  File "/usr/users/ga002/soranzon/software/galaxyproject_galaxy/lib/galaxy/webapps/galaxy/api/workflows.py", line 1456, in export_invocation_bco
    bco = self._deprecated_generate_bco(trans, invocation_id, merge_history_metadata)
  File "/usr/users/ga002/soranzon/software/galaxyproject_galaxy/lib/galaxy/webapps/galaxy/api/workflows.py", line 1509, in _deprecated_generate_bco
    return self.invocations_service.deprecated_generate_invocation_bco(trans, invocation_id, export_options)
  File "/usr/users/ga002/soranzon/software/galaxyproject_galaxy/lib/galaxy/webapps/galaxy/services/invocations.py", line 229, in deprecated_generate_invocation_bco
    export_target.seek(0)
  File "/usr/users/ga002/soranzon/software/galaxyproject_galaxy/lib/galaxy/model/store/__init__.py", line 2146, in __exit__
    self._finalize()
  File "/usr/users/ga002/soranzon/software/galaxyproject_galaxy/lib/galaxy/model/store/__init__.py", line 2274, in _finalize
    core_biocompute_object, object_id = self._core_biocompute_object_and_object_id()
  File "/usr/users/ga002/soranzon/software/galaxyproject_galaxy/lib/galaxy/model/store/__init__.py", line 2362, in _core_biocompute_object_and_object_id
    pipeline_step = PipelineStep(
  File "pydantic/main.py", line 342, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 1 validation error for PipelineStep
name
  none is not an allowed value (type=type_error.none.not_allowed)
```

mvdbeek commented on lib/galaxy/tools/duplicate_file_to_collection.xml:
```diff
@@ -9,9 +9,9 @@
-        <param type="text" name="element_identifier" label="Element identifier" help="Each element in the output collection will be assigned this identifier, with an integer appended." />
+        <param name="input" type="data" label="Input Dataset" help="A dataset which will be duplicated."/>
+        <param name="number" type="integer" value="" label="Size of output collection" help="Number of times to duplicate the input dataset."/>
+        <param name="element_identifier" type="text" label="Element identifier" help="Each element in the output collection will be assigned this identifier, with an integer appended." />
```
type_source doesn't make sense either, should be list, i think.

nsoranzo replied:
Thanks, fixed that and a few more issues, now the tool works!

nsoranzo committed:
More fixes for ``__DUPLICATE_FILE_TO_COLLECTION__`` tool

- Fix output collection type
- Fix test output collection name
- Number elements identifiers starting from 1
- Add new HDAs to history, fix traceback:

  ```
  ERROR    galaxy.tools:__init__.py:1922 Exception caught while attempting to execute tool with id '__DUPLICATE_FILE_TO_COLLECTION__':
  Traceback (most recent call last):
    File "/usr/users/ga002/soranzon/software/nsoranzo_galaxy/lib/galaxy/tools/__init__.py", line 1900, in handle_single_execution
      rval = self.execute(
    File "/usr/users/ga002/soranzon/software/nsoranzo_galaxy/lib/galaxy/tools/__init__.py", line 1995, in execute
      return self.tool_action.execute(
    File "/usr/users/ga002/soranzon/software/nsoranzo_galaxy/lib/galaxy/tools/actions/model_operations.py", line 83, in execute
      self._produce_outputs(
    File "/usr/users/ga002/soranzon/software/nsoranzo_galaxy/lib/galaxy/tools/actions/model_operations.py", line 108, in _produce_outputs
      tool.produce_outputs(
    File "/usr/users/ga002/soranzon/software/nsoranzo_galaxy/lib/galaxy/tools/__init__.py", line 3685, in produce_outputs
      output_collections.create_collection(
    File "/usr/users/ga002/soranzon/software/nsoranzo_galaxy/lib/galaxy/tools/actions/__init__.py", line 1014, in create_collection
      check_elements(elements)
    File "/usr/users/ga002/soranzon/software/nsoranzo_galaxy/lib/galaxy/tools/actions/__init__.py", line 1005, in check_elements
      assert value.history is not None or value.history_id is not None
  AssertionError
  ```
